ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php

```diff
@@ -0,0 +1,200 @@
+                'currentPage' => $pagination->currentPage(),
+                'lastPage' => $pagination->lastPage(),
+            ],
+        ];
+    }
+
+    /**
+     * Apply filters to the query.
+     *
+     * @param Request $request
+     * @param Builder $query
+     *
+     * @return void
+     */
+    private function filters(Request $request, Builder $query): void
+    {
+        if ($request->has('userId')) {
+            $query->where('user_id', $request->get('userId'));
+        }
+
+        if ($request->has('status')) {
+            $query->where('case_status', $request->get('status'));
+        }
+
+        $this->search($request, $query);
+        $this->filterBy($request, $query);
+        $this->sortBy($request, $query);
+    }
+
+    /**
+     * Sort the query.
+     *
+     * @param Request $request: Query parameter format: sortBy=field:asc,field2:desc,...
+     * @param Builder $query
+     *
+     * @return void
+     */
+    private function sortBy(Request $request, Builder $query): void
+    {
+        $sort = explode(',', $request->get('sortBy'));
+
+        foreach ($sort as $value) {
+            if (!preg_match('/^[a-zA-Z_]+:(asc|desc)$/', $value)) {
+                continue;
+            }
+
+            $sort = explode(':', $value);
+            $field = $sort[0];
+            $order = $sort[1] ?? 'asc';
+
+            if (in_array($field, $this->sortableFields)) {
+                $query->orderBy($field, $order);
+            }
+        }
+    }
+
+    /**
+     * Filter the query.
+     *
+     * @param Request $request: Query parameter format: filterBy[field]=value&filterBy[field2]=value2&...
+     * @param Builder $query
+     * @param array $dateFields List of date fields in current model
+     *
+     * @return void
+     */
+    private function filterBy(Request $request, Builder $query): void
+    {
+        if ($request->has('filterBy')) {
+            $filterByValue = $request->get('filterBy');
+
+            foreach ($filterByValue as $key => $value) {
+                if (!in_array($key, $this->filterableFields)) {
+                    continue;
+                }
+
+                if (in_array($key, $this->dateFields)) {
+                    $query->whereDate($key, $value);
+                    continue;
+                }
+
+                $query->where($key, $value);
+            }
+        }
+    }
+
+    /**
+     * Search by case number or case title.
+
+     * @param Request $request: Query parameter format: search=keyword
+     * @param Builder $query
+     *
+     * @return void
+     */
+    private function search(Request $request, Builder $query): void
+    {
+        if ($request->has('search')) {
+            $search = $request->get('search');
+
+            $query->where(function ($q) use ($search) {
+                foreach ($this->searchableFields as $field) {
+                    if ($field === 'case_number') {
+                        $q->orWhere($field, 'like', "%$search%");
```

Comment:
Consider to change this to equal operation, and add that detail to the architecture document to search by case_number

Reply:
It was fixed